JSQMessagesViewController/Controllers/JSQMessagesViewController.m

```diff
@@ -402,14 +402,18 @@ - (void)scrollToIndexPath:(NSIndexPath *)indexPath animated:(BOOL)animated
     }
 
     CGFloat collectionViewContentHeight = [self.collectionView.collectionViewLayout collectionViewContentSize].height;
-    BOOL isContentTooSmall = (collectionViewContentHeight < CGRectGetHeight(self.collectionView.bounds));
-
+    CGFloat collectionViewHeight = CGRectGetHeight(self.collectionView.bounds);
+    BOOL isContentTooSmall = (collectionViewContentHeight < collectionViewHeight + 10.f);
+    
     if (isContentTooSmall) {
         //  workaround for the first few messages not scrolling
         //  when the collection view content size is too small, `scrollToItemAtIndexPath:` doesn't work properly
         //  this seems to be a UIKit bug, see #256 on GitHub
-        [self.collectionView scrollRectToVisible:CGRectMake(0.0, collectionViewContentHeight - 1.0f, 1.0f, 1.0f)
-                                        animated:animated];
+        
+        CGFloat contentWindowHeight = collectionViewHeight - self.collectionView.contentInset.top - self.collectionView.contentInset.bottom;
+        CGFloat offsetHeight = MAX(0.f, collectionViewContentHeight - contentWindowHeight) - self.collectionView.contentInset.top;
```

Comment:
Not exactly twice, only if content passed visible portion of scrolling view. 
in `contentWindowHeight` it subtracting for get actual value of visible height of scroll view, 
second time, in `offsetHeight`, it's subtracted as part of `contentOffset` - to get correct offset value